public/components/visualization/source_selector.tsx

```diff
@@ -114,9 +114,12 @@ export const SourceSelector = ({
        * If not configured, disable the corresponding index pattern from the selection list
        */
       Object.keys(dataSourceIdToIndexPatternIds).forEach(async (key) => {
-        const res = await assistantService.client.agentConfigExists(TEXT2VEGA_AGENT_CONFIG_ID, {
-          dataSourceId: key !== 'DEFAULT' ? key : undefined,
-        });
+        const res = await assistantService.client.agentConfigExists(
+          TEXT2VEGA_RULE_BASED_AGENT_CONFIG_ID,
+          {
+            dataSourceId: key !== 'DEFAULT' ? key : undefined,
+          }
+        );
```

Comment:
I am not sure whether the logic should be both `TEXT2VEGA_RULE_BASED_AGENT_CONFIG_ID` and `TEXT2VEGA_AGENT_CONFIG_ID` are not configured then disable the index pattern?

Reply:
Good question, I think both should be configured, or the index pattern should be disabled.
If there is only one configured, it's still problematic, for example, only `TEXT2VEGA_RULE_BASED_AGENT_CONFIG_ID` is configured, but user's input contains instructions, it will use instruction based agent.

Reply:
agree, let's do the check for both.

Reply:
@Hailong-am Updated, the new implementation will check all required agents includes text to ppl, please help to take another look, thanks!